Indexing/SubstitutionTree.cpp

```diff
@@ -50,14 +51,9 @@ using namespace Indexing;
  * Initialise the substitution tree.
  * @since 16/08/2008 flight Sydney-San Francisco
  */
-SubstitutionTree::SubstitutionTree(bool useC, bool rfSubs)
+SubstitutionTree::SubstitutionTree()
```

Comment:
I'm not quite sure what you mean... There's still the fields `_nextVar` and `_root` that need to be initialized in the constructor.

Reply:
ahh i see.. you mean writing
```
class SubstitutionTree {
...
    int _nextVar = 0;
    Node* _root = nullptr;
...
};
```
I wasn't aware you could do that in c++. I thought this was a java-only thing. 
I don't care we can do it either way but i never saw it like this around the codebase. But I reckon I'm quite in favor of using it if we can actually.

Reply:
changed it

Reply:
I think it's a relatively recent thing. With older standards the fields need to static? and const? or something. 

Also a bit hesitant to change the coding style, but since the initialization is with the declaration, I can see how it could bite us and it does feel like an improvement (in this way things are done).

Reply:
Right - I wasn't aware until recently either, but I think it's been available since C++11. Generally in favour, especially if we can remove a constructor altogether. Thanks for the change!